### ISSUE TITLE
Allow for viewing CloudFront's CloudWatch Logs

### ISSUE THREAD

**Issue by Chris1904** [aws/aws-toolkit-vscode#1886]
**Is your feature request related to a problem? Please describe.**

It is not related to a problem. In fact, the plugin is working as expected but when working with Lambda@edge functions, the relating CloudWatch log files may be stored in different regions. 

**Describe the solution you'd like**

It would be incredibly useful for debugging reasons to have "CloudFront Logs" which could be structured similarly to "CloudWatch Logs" but it simply would be an aggregation of all of the CloudFront logs through all regions that contain CloudFront logs compared to the one your AWS account has currently selected.

**Describe alternatives you've considered**

The only alternative at the moment is searching for those log files manually. This flow requires you to leave the editor and it would be nice if you could stay completely inside it.

**Additional context**

![image](https://user-images.githubusercontent.com/6228425/126914694-2a87a59c-9071-42ea-a146-fea9d08ef9fb.png)

(just adding the screenshot for reference of what section I am addressing)

In terms of how I would structure the UI aspect, I likely would choose to nest the CloudFront logs as follows: 

CloudWatch Logs > CloudFront Logs > Regions

Would love to hear if others would find this a useful and logical addition.

**Comment by JadenSimon** [comment]
Interesting proposal. So you're wanting a new node _inside_ the CloudWatch Logs node that is effectively the same across regions?

I'm not too familiar with CloudFront logs, but I thought they were normally sent to an S3 bucket for standard logging. If that's the case, then maybe we could add a way to list CloudFront logs from a designated bucket.

Some more context would be very helpful in determining how we should proceed, though this initial proposal is great nonetheless.

**Comment by Chris1904** [comment]
Thanks for your kind response. And totally, let me provide some more context.

We utilize an Origin Request Lambda@edge function to apply some business logic prior to returning the image from the "Origin server."
 
![image](https://user-images.githubusercontent.com/6228425/127043881-29d08a11-a97d-4b12-a86d-4357ed549714.png)

Based on the geographical location where the request came from, it is used to determine in which region to store the associated Lambda@edge logs. 

For example:

My default region for my AWS account is `us-east-1` which means the CloudWatch Logs I am seeing are `us-east-1`-logs as

<img width="1021" alt="Screen Shot 2021-07-26 at 12 05 41 PM" src="https://user-images.githubusercontent.com/6228425/127044498-03a70d48-7a1f-4fd2-baec-c8355374e023.png">

As you can see, those 2 `us-east-1` log files match what there is shown inside CloudWatch.

![image](https://user-images.githubusercontent.com/6228425/127044364-dc0559bb-e685-4594-a74d-cd7982a0583d.png)

Now, the tricky part when debugging these Lambda@edge functions is that you have to figure out which region has logs for that associated edge function. 

As an example, CloudFront utilizes the `us-west-2` location for my personal requests which means that I could not find these log files inside Code Explorer.

![image](https://user-images.githubusercontent.com/6228425/127045079-05aeb7e2-cc05-4f7a-ab05-e9db0dbd6716.png)


To sum this all up, it would be great if we could add the CloudFront associated logs for all regions into the "CloudWatch Logs" because it would drastically simplify and streamline the Developer's Experience because you could then fully live inside VS Code without having the mental context switch each time you are debugging your edge functions.


PS: since CloudFront is a global service, it likely could make sense to structure it in this type of way as well:

![image](https://user-images.githubusercontent.com/6228425/127046185-73aac77d-972b-48e7-90e3-0af477846ada.png)

**Comment by JadenSimon** [comment]
Oh wow thank you for the background info, it's very helpful! I think I understand the feature request much better now. In effect, the goal here is to organize CloudWatch logs in a logical way. 

Would it be correct to say that in your specific use case, their exists multiple log groups across multiple regions with the same name? If that's the case, perhaps the generic solution here would be to have a feature for coalescing same-named log groups from multiple regions, displaying them as a 'grouped' node (global does make sense here). Then each sub-node would be a different region just like you were describing.

Presumably this desire to organize log groups isn't unique to CloudFront, which is why I'm trying to frame this in a more abstract way.

**Comment by Chris1904** [comment]
> Would it be correct to say that in your specific use case, their exists multiple log groups across multiple regions with the same name?

Yes, you are completely right. CloudFront creates these log groups and their name is same across all regions.

> Presumably this desire to organize log groups isn't unique to CloudFront, which is why I'm trying to frame this in a more abstract way.

This totally makes sense as well and you are right that this concept may be applied to other "global" services. 🙂
